swap_meet/vendor.py

```diff
@@ -1,2 +1,112 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory: # or if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    
+    def get_by_category(self, category1):
+        category_list = []
+        for item in self.inventory:
+            if category1 == item.category:
+                category_list.append(item) #Item(category="clothing")
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            exchange_item = self.remove(my_item)
```

Comment:
you made this `exchange_item` variable but didn't need to. You can just say `self.remove(my_item)` and on line 28 say `friend.add(my_item)`